---
title: "ddPCR"
author: "Emily Lacroix"
date: "25 APR 2024"
output: 
  github_document:
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Setup

## Load libraries
```{r echo=TRUE, results ='hide', warning=FALSE, message=FALSE}
library(car)
library(multcompView)
library(FSA)
library(outliers)
library(tidyverse)
library(readxl)
library(ggpubr)
library(ggfortify)
library(scales)
library(rstatix)
library(rcompanion)
```

## Functions

```{r}
tri.to.squ<-function(x)
{
  rn <- row.names(x)
  cn <- colnames(x)
  an <- unique(c(cn,rn))
  myval <- x[!is.na(x)]
  mymat <- 
    matrix(1, nrow = length(an), ncol = length(an), dimnames = list(an,an))
  for(ext in 1:length(cn))
  {
    for(int in 1:length(rn))
    {
      if(is.na(x[row.names(x) == rn[int], colnames(x) == cn[ext]])) next
      mymat[row.names(mymat) == rn[int], colnames(mymat) == cn[ext]] <- 
        x[row.names(x) == rn[int], colnames(x) == cn[ext]]
      mymat[row.names(mymat) == cn[ext], colnames(mymat) == rn[int]] <- 
        x[row.names(x) == rn[int], colnames(x) == cn[ext]]
    }
  }
  return(mymat)
}


```

## Files

```{r eval=FALSE}

grav_data_file <- "intact_core_grav.csv"

extract_data_file <- "dna_extraction.csv"

all_data_file <- "AllData_FINAL.xlsx"

mcrA_file <- "mcra_data.xlsx"

glta_file1 <- "glta_SHI_28oct_manual.csv"

glta_file2 <- "lacroix_glta_02nov_manual.csv"

nirK_file1 <- "EML_nirK_SHI_samples_16jan_auto.csv" #note - this is the autointegration assignment from QuantaSoft software

nirS_file1 <-  "EML_nirS_SHI_samples_21jan_MANUAL_6000.csv" #note - this is the manual integration assignment, with threshold = 6000

dsrab_file <- "EML_dsrAB_SHI_samples_15mar2022_auto.csv"



```

```{r eval=TRUE, echo=FALSE}

grav_data_file <- "/Users/elacroi3/Documents/Research/SHI/Data/DataCompilation/intact_core_grav.csv"

extract_data_file <- "/Users/elacroi3/Documents/Research/SHI/Data/DataCompilation/dna_extraction.csv"

all_data_file <- "/Users/elacroi3/Documents/Research/SHI/Data/DataCompilation/AllData_FINAL.xlsx"

mcrA_file <- "/Users/elacroi3/Documents/Research/SHI/Data/DataCompilation/mcra_data.xlsx"

glta_file1 <- "/Users/elacroi3/Documents/Research/SHI/Data/Microbial/ddPCR/gltA/glta_SHI_28oct_manual.csv"

glta_file2 <- "/Users/elacroi3/Documents/Research/SHI/Data/Microbial/ddPCR/gltA/lacroix_glta_02nov_manual.csv"

nirK_file1 <- "/Users/elacroi3/Documents/Research/SHI/Data/Microbial/ddPCR/nirK/EML_nirK_SHI_samples_16jan_auto.csv" #note - this is the autointegration assignment from QuantaSoft software

nirS_file1 <-  "/Users/elacroi3/Documents/Research/SHI/Data/Microbial/ddPCR/nirS/EML_nirS_SHI_samples_21jan_MANUAL_6000.csv" #note - this is the manual integration assignment, with threshold = 6000

dsrab_file <-  "/Users/elacroi3/Documents/Research/SHI/Data/Microbial/ddPCR/dsrAB/EML_dsrAB_SHI_samples_15mar2022_auto.csv"

#output_file <-  "~/Google Drive/My Drive/BoxMigration/Stanford (elacroix@stanford.edu)/Research/SHI/Data/DataCompilation/dna_ddpcr.csv"

#total_anaerobe_file <-  "~/Google Drive/My Drive/BoxMigration/Stanford (elacroix@stanford.edu)/Research/SHI/Data/DataCompilation/total_anaerobe.csv"

```


## Figure theme
```{r}
my_theme <- function(base_size = 13, base_family = ""){ ## Control base font face and size. use `rel()` for relative font size.
  theme_bw(base_size = base_size, base_family = base_family) %+replace%
    theme(
      panel.border = element_rect(colour = "black", fill = "transparent"),
      panel.background  = element_blank(),
      panel.grid = element_blank(),
      strip.background = element_blank(),
      legend.position="top",
      legend.title = element_blank(),
      strip.text = element_text(size = 11),
      strip.text.y = element_text(size = 11)
    )
}

```


## Labels
```{r}
site_labels <- 
  c(
    "CREC" = "Carrington, ND",
    "WO" = "Wooster, OH",
    "GR" = "Novelty, MO",
    "SM" = "Crossville, AL"
  )
```

## Constants

```{r}

rxn_dilution_factor <- 25

```


# Import and combine data

## Extraction data

```{r warning=FALSE, message=FALSE}
dna_extraction <- 
  extract_data_file %>% 
  read_csv %>% 
  mutate(
    across(landscape_position, ~na_if(., "NA"))
  ) %>% 
  select(
    site,
    till,
    amend,
    landscape_position,
    rep = fieldrep,
    mass_g,
    nano_260_280,
    nano_260_230,
    qubit_ng_ul,
    qubit_ng_mL
  )

dna_extraction
```


## Soil Moisture data

```{r warning=FALSE, message=FALSE}

bd_est <- 
  all_data_file %>% 
  read_xlsx(sheet = "gravimetric", na = "NA") %>% 
  group_by(
    site, 
    till, 
    amend, 
    field_rep, 
    landscape_position,
    vwc_probe
  ) %>% 
  summarise(
    avg_bd = mean(bd, na.rm = TRUE),
    avg_vwc = mean(vwc, na.rm = TRUE),
    avg_gwc = mean(gwc, na.rm = TRUE)
  ) %>% 
  select(    
    site, 
    till, 
    amend, 
    field_rep, 
    landscape_position,
    vwc_probe,
    avg_bd,
    avg_vwc,
    avg_gwc
  ) %>% 
  ungroup()




```


## Import ddPCR results

### mcrA

This code:

* reads in the mcra data 
* designates "No Calls" to 0 (confirmed visually)
* selects relevant data fields 
* omits data with fewer than 10,000 droplets


```{r warning=FALSE, message=FALSE}
mcrA_data <- 
  mcrA_file %>% 
  read_xlsx(sheet = "ddPCR") %>% 
  mutate(across(landscape_position, ~na_if(., "NA"))) %>% 
  mutate(
    target_gene = "mcrA",
    across(c(conc, conc_min, conc_max), as.numeric),
    across(conc, ~if_else(is.na(.), 0, .)) #treat NAs like 0s
  ) %>% 
  select(
    -c(
      target_type, 
      status, 
      positives, 
      negatives, 
      copies_per_20uL
    )
  ) %>% 
  rename(rep = field_rep) %>% 
  filter(total_droplets > 10000)


mcrA_data %>% arrange(site, till, amend, rep, landscape_position)


```

### gltA

This code:

* designates the PCR template dilution
* reads in the glta data 
* selects relevant data fields
* omits data with fewer than 10,000 droplets


```{r warning=FALSE, message=FALSE}
glta_df <- 10

glta_data <-
  glta_file2 %>% 
  read_csv() %>% 
  mutate(across(c(Concentration), as.numeric)) %>% 
  bind_rows(glta_file1 %>% read_csv()) %>% 
  select(
    sample_name = Sample,
    target_type = TargetType,
    status = Status,
    conc = Concentration, 
    copies_per_20uL = CopiesPer20uLWell,
    conc_min = PoissonConfMin,
    conc_max = PoissonConfMax,
    positives = Positives,
    negatives = Negatives,
    total_droplets = AcceptedDroplets
  ) %>% 
  separate(sample_name, into = c("site", "till", "amend", "rep")) %>% 
  mutate(
    landscape_position = if_else(site == "GR", amend, NA_character_),
    across(amend, ~if_else(site == "GR", "U", .)),
    across(c(conc, rep), as.numeric),
    target_gene = "gltA"
  ) %>% 
  filter(site %in% c("SM", "CREC", "GR", "WO")) %>% 
  mutate(
    across(c(conc, conc_min, conc_max), ~ . * glta_df)
  ) %>% 
  select(
    -c(
      target_type, 
      status, 
      positives, 
      negatives, 
      copies_per_20uL, 
      )
  ) %>% 
  filter(total_droplets > 10000)

```

### nirK

This code:

* reads in the nirK data 
* selects relevant data fields 
* corrects for template dilution
* omits data with fewer than 10,000 droplets


```{r warning=FALSE, message=FALSE}
nirk_data <-
  nirK_file1 %>% 
  read_csv() %>% 
  select(
    sample_name = Sample,
    target_type = TargetType,
    status = Status,
    conc = Concentration, 
    copies_per_20uL = CopiesPer20uLWell,
    conc_min = PoissonConfMin,
    conc_max = PoissonConfMax,
    positives = Positives,
    negatives = Negatives,
    total_droplets = AcceptedDroplets
  ) %>% 
  separate(
    sample_name, 
    into = c("site", "till", "amend", "rep"),
    sep = c("-")
  ) %>% 
  mutate(
    dilution = str_extract(rep, "(?<=:)\\d+"),
    across(rep, ~str_extract(., "^\\d")),
    landscape_position = if_else(site == "GR", amend, NA_character_),
    across(amend, ~if_else(site == "GR", "U", .)),
    across(c(conc, dilution, rep), as.numeric),
    target_gene = "nirK"
  ) %>% 
  filter(site %in% c("SM", "CREC", "GR", "WO")) %>% 
  mutate(
    across(c(conc, conc_min, conc_max), ~ . * dilution)
  ) %>% 
  select(
    -c(
      target_type, 
      status, 
      positives, 
      negatives, 
      copies_per_20uL, 
      dilution
      )
  ) %>% 
  filter(total_droplets > 10000)


```


### nirS

This code:

* reads in the nirs data 
* selects relevant data fields and renames to be nirs specific
* corrects for template dilution
* omits data with fewer than 10,000 droplets


```{r warning=FALSE, message=FALSE}
nirs_data <-
  nirS_file1 %>%
  read_csv() %>%
  select(
    sample_name = Sample,
    target_type = TargetType,
    status = Status,
    conc = Concentration,
    copies_per_20uL = CopiesPer20uLWell,
    conc_min = PoissonConfMin,
    conc_max = PoissonConfMax,
    positives = Positives,
    negatives = Negatives,
    total_droplets = AcceptedDroplets
  ) %>%
  separate(
    sample_name,
    into = c("site", "till", "amend", "rep"),
    sep = c("-")
  ) %>% 
  mutate(
    dilution = str_extract(rep, "(?<=:)\\d+"),
    across(rep, ~str_extract(., "^\\d")),
    landscape_position = if_else(site == "GR", amend, NA_character_),
    across(amend, ~if_else(site == "GR", "U", .)),
    across(c(conc, dilution, rep), as.numeric),
    target_gene = "nirS"
  ) %>% 
  filter(site %in% c("SM", "CREC", "GR", "WO")) %>%
  mutate(
    across(c(conc, conc_min, conc_max), ~ . * dilution)
  ) %>%
  select(
    -c(
      target_type,
      status,
      positives,
      negatives,
      copies_per_20uL,
      dilution
      )
  ) %>% 
  filter(total_droplets > 10000)



```


### dsrAB

This code:

* specifies the dilution factor for the DNA template
* reads in the dsrAB data 
* selects relevant data fields 
* corrects for template dilution
* omits data with fewer than 10,000 droplets


```{r warning=FALSE, message=FALSE}

dsrab_dilution_factor <- 50

dsrab_data <-
  dsrab_file %>%
  read_csv() %>%
  select(
    sample_name = Sample,
    target_type = TargetType,
    status = Status,
    conc = Concentration,
    copies_per_20uL = CopiesPer20uLWell,
    conc_min = PoissonConfMin,
    conc_max = PoissonConfMax,
    positives = Positives,
    negatives = Negatives,
    total_droplets = AcceptedDroplets
  ) %>%
  separate(
    sample_name,
    into = c("site", "till", "amend", "rep"),
    sep = c("-")
  ) %>% 
  mutate(
    landscape_position = if_else(site == "GR", amend, NA_character_),
    across(amend, ~if_else(site == "GR", "U", .)),
    across(c(conc, rep), as.numeric)
  ) %>% 
  filter(site %in% c("SM", "CREC", "GR", "WO")) %>%
  mutate(
    across(c(conc, conc_min, conc_max), ~ . * dsrab_dilution_factor),
    target_gene = "dsrAB"
  ) %>% 
  select(
    -c(
      target_type,
      status,
      positives,
      negatives,
      copies_per_20uL
      )
  ) %>% 
  filter(total_droplets > 10000)

```


## Combine data

This code:

* combines all the target gene tibbles into one
* corrects concentration values to account for dilution within PCR 
* calculates target gene abundance normalized to 16S abundance
* calculates copies per g soil extracted

```{r}

all_ddpcr_data_w_outliers <- 
  bind_rows(mcrA_data, glta_data, dsrab_data, nirk_data, nirs_data) %>% 
  left_join(
    dna_extraction,
    by =
      c("site", "till", "amend", "landscape_position", "rep")
  ) %>% 
  left_join(
    bd_est,
    by =
      c("site", "till", "amend", "landscape_position", "rep" = "field_rep")
  ) %>% 
  mutate(
    across(
      c(conc, 
        conc_min, 
        conc_max, 
      ), ~ . * rxn_dilution_factor
    ),
    g_h2o = mass_g / (1 / avg_gwc + 1),
    mass_dry_soil = mass_g - g_h2o,
    copies_per_g = conc * 100 / mass_dry_soil
  ) 


```


## Check for outliers

### Total anaerobes

```{r message=FALSE}
total_anaerobe_w_outliers <-
  all_ddpcr_data_w_outliers %>% 
  group_by(site, till, amend, landscape_position, target_gene, rep) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
  ) %>% 
  group_by(site, till, amend, landscape_position, rep) %>% 
  summarise(
    anaerobe_copies_per_g = sum(avg_copies_per_g),
    count = n() #missing genes for GR-NT-F-1 and GR-NT-F-3 due to droplet generation issues
  ) %>% 
  filter(count == 5)

outliers_total <- grubbs.test(total_anaerobe_w_outliers$anaerobe_copies_per_g)

outliers_total
```
Grubb's Test shows that the highest total anaerobe value is a statistical outlier. 


```{r}
total_anaerobe <- 
  total_anaerobe_w_outliers %>% 
  filter(!(site == "WO" & till == "UN" & rep == 2))

all_ddpcr_data <- 
  all_ddpcr_data_w_outliers %>% 
  filter(!(site == "WO" & till == "UN" & rep == 2))
  
outliers <- grubbs.test(total_anaerobe$anaerobe_copies_per_g)

```


# Absolute Abundance: Unamended, effect of tillage across sites

## Figure 2b 

```{r warning=FALSE, message=FALSE}

all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n()),
    count = n()
  ) %>% 
  filter(site != "GR", amend == "U", till %in% c("CT", "NT", "UN")) %>% 
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  ggplot(aes(y = avg_copies_per_g, x = till, fill = target_gene)) + 
  geom_col(position = "dodge", color = "black") +
  geom_errorbar(
    aes(
      ymin = avg_copies_per_g,
      ymax = avg_copies_per_g + se_copies_per_g
    ),
    position = "dodge"
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(
    labels = scientific, 
    n.breaks = 4,
    expand = expand_scale(mult = c(0, 0.25))
  ) +
  facet_grid(
    cols = vars(site), 
    rows = vars(target_gene), 
    scales = "free",
    labeller = labeller(site = site_labels)
  ) + 
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 1.1,
    legend.position = "none",
    strip.text.y = element_text(face = "italic"),
    strip.text.x = element_text(size = 10)
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )

```

## Statistical tests - reported in Supplementary Table S2

### Statistics - nirK

#### nirK - CREC

**Normality:** no single transformation makes all 3-4 groups normal
```{r}

all_ddpcr_data %>% 
  filter(site == "CREC", amend == "U",target_gene == "nirK") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```
**Variances:** equal variances

```{r warning = FALSE}

leveneTest( 
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", amend == "U",target_gene == "nirK") 
) 

```

**Statistical test:** Kruskal-Wallis test

```{r}

kruskal.test(
  copies_per_g ~ till, 
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", amend == "U",target_gene == "nirK")
)

```

#### nirK - SM

**Normality:** all normal
```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "SM", amend == "U",target_gene == "nirK") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** not equal

```{r}

bartlett.test( 
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "nirK") 
) 

```

**Statistics:** Welch's ANOVA 

```{r}

oneway.test(
  copies_per_g ~ till, 
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U", target_gene == "nirK"),
  var.equal = FALSE
)


games_howell_test(
  copies_per_g ~ till, 
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U", target_gene == "nirK")
)

```


#### nirK -  WO

**Normality:** Normal
```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "WO", amend == "U",target_gene == "nirK") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** equal

```{r warning=FALSE}
leveneTest( 
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "nirK") 
) 

```

**Statistics:** ANOVA

```{r}

wo_nirk_aov <- 
  aov(
      copies_per_g ~ till,
      data = 
        all_ddpcr_data %>% 
        filter(site == "WO", amend == "U",target_gene == "nirK") 
)


summary(wo_nirk_aov)

tukey_wo_nirk <- TukeyHSD(wo_nirk_aov)

tukey_wo_nirk


```


### Statistics - nirS



#### nirS - CREC

**Normality:** All normal.

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "CREC", amend == "U",target_gene == "nirS") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** Equal

```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", amend == "U",target_gene == "nirS") 
)

```

**Statistics:** ANOVA

```{r}
aov(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", amend == "U", target_gene == "nirS") 
) %>% 
  summary()
```

#### nirS - SM

**Normality:** Normal after log transformation

```{r message=FALSE}
all_ddpcr_data %>% 
  filter(site == "SM", amend == "U",target_gene == "nirS") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(log(copies_per_g))$p.value
  ) %>%
  arrange(site, normality_p)
```

**Variances:** equal
```{r}
bartlett.test(
  log(copies_per_g) ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "nirS") 
)
```

**Statistics:** ANOVA

```{r}

aov(
  log(copies_per_g) ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "nirS") 
) %>% summary()

```

#### nirS - WO

**Normality:** All normal

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "WO", amend == "U",target_gene == "nirS") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```


**Variances:**
```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "nirS") 
)

```


**Statistics:** ANOVA

```{r}
aov(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "nirS") 
) %>% summary()
```

### Statistics gltA

#### gltA - CREC

**Normality:** all approximately normal 
```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "CREC", amend == "U",target_gene == "gltA") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** Equal variances

```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", amend == "U",target_gene == "gltA") 
)

```

**Statistics:** ANOVA

```{r}

crec_glta_anova <- 
  aov(
    copies_per_g ~ till,
    data = 
      all_ddpcr_data %>% 
      filter(site == "CREC", amend == "U",target_gene == "gltA") 
  ) 

crec_glta_anova %>% summary()

tukey_crec_glta <- TukeyHSD(crec_glta_anova)

multcompLetters4(crec_glta_anova, tukey_crec_glta)
```

#### gltA - SM

**Normality:** all approximately normal 
```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "SM", amend == "U",target_gene == "gltA") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```


**Variances:** equal variances

```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "gltA") 
)

```


**Statistics:** ANOVA

```{r}

aov(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "gltA") 
) %>% 
  summary()

```


#### gltA - WO

**Normality:** all approximately normal 

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "WO", amend == "U",target_gene == "gltA") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** Barely equal variances

```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "gltA") 
)

```


**Statistics:**  ANOVA

```{r}

aov(
  copies_per_g ~ till, 
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "gltA")
) %>% summary()

```

### Statistics - dsrAB

#### dsrAB - CREC

**Normality:** all approximately normal 
```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "CREC", amend == "U", target_gene == "dsrAB") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** equal
```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", amend == "U",target_gene == "dsrAB") 
)

```

**Statistics:** ANOVA

```{r}

crec_dsrab_anova <- 
  aov(
    copies_per_g ~ till,
    data = 
      all_ddpcr_data %>% 
      filter(site == "CREC", amend == "U",target_gene == "dsrAB") 
  ) 

crec_dsrab_anova %>% summary()

tukey_crec_dsrab <- TukeyHSD(crec_dsrab_anova)

multcompLetters4(crec_dsrab_anova, tukey_crec_dsrab)

```


#### dsrAB - SM


**Normality:** all approximately normal 
```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "SM", amend == "U", target_gene == "dsrAB") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** Unequal

```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U", target_gene == "dsrAB") 
)

```

**Statistics:** Welch's ANOVA

```{r}

oneway.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "dsrAB") 
)

```


#### dsrAB - WO

**Normality:** Non-normal, even after transformation attempts
```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "WO", amend == "U", target_gene == "dsrAB") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(site, normality_p)

```

**Variances:** approximately equal

```{r warning=FALSE}

leveneTest(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "dsrAB") 
)

```

**Statistics:** Kruskal-Wallis

```{r}

kruskal.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "dsrAB") 
)

dunnTest(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "dsrAB"),
  method = "bh"
)

```


### Statistics - mcrA

#### mcrA - CREC

**Normality:** All approximately normal

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "CREC", amend == "U",target_gene == "mcrA") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(normality_p)

```


**Variances:** Equal

```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", amend == "U",target_gene == "mcrA") 
)

```


**Statistics:** ANOVA

```{r}

crec_mcra_anova <- 
  aov(
    copies_per_g ~ till,
    data = 
      all_ddpcr_data %>% 
      filter(site == "CREC", amend == "U",target_gene == "mcrA") 
  ) 

crec_mcra_anova %>% summary()

tukey_crec_mcra <- TukeyHSD(crec_mcra_anova)

multcompLetters4(crec_mcra_anova, tukey_crec_mcra)

```

#### mcrA - SM

**Normality:** All approximately normal

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "SM", amend == "U",target_gene == "mcrA") %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(normality_p)

```

**Variances:** Unequal

```{r}

bartlett.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "mcrA") 
)

```

**Statistics:** Welch's ANOVA

```{r}

oneway.test(
  copies_per_g ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "SM", amend == "U",target_gene == "mcrA"),
  var.equal = FALSE #this is the default setting 
)


```


#### mcrA - WO

**Normality:** Normal after log transformation

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(site == "WO", amend == "U", target_gene == "mcrA") %>% 
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(log(copies_per_g))$p.value
  ) %>%
  arrange(normality_p)

```

**Variances:** Equal

```{r}

bartlett.test(
  log(copies_per_g) ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "mcrA") 
)

```

**Statistics:** ANOVA

```{r}
aov(
  log(copies_per_g) ~ till,
  data = 
    all_ddpcr_data %>% 
    filter(site == "WO", amend == "U",target_gene == "mcrA") 
) %>% summary()


```

## Figure 2a - stacked bars

```{r message=FALSE}

total_anaerobe <-
  all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene, rep) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
  ) %>% 
  group_by(site, till, amend, landscape_position, rep) %>% 
  summarise(
    anaerobe_copies_per_g = sum(avg_copies_per_g),
    count = n() #Missing genes for GR-NT-F-1 and GR-NT-F-3 due to droplet generation issues
  ) %>% 
  filter(count == 5)

errors <-
  all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n()),
  ) %>% 
  group_by(site, till, amend, landscape_position) %>% 
  summarise(
    sum_se = sum(se_copies_per_g),
    sum_copies = sum(avg_copies_per_g),
    count = n()
  ) 


all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n()),
  ) %>% 
  filter(site != "GR", amend == "U", till %in% c("CT", "NT", "UN")) %>%
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  left_join(errors, by = c("site", "till", "amend", "landscape_position")) %>% 
  ggplot(aes(x = till, fill = target_gene)) + 
  geom_col(aes(y = avg_copies_per_g), position = "stack", color = "black") +
  geom_errorbar(
    aes(y = sum_copies, ymin = sum_copies, ymax = sum_copies + sum_se)
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(
    labels = scientific,
    expand = expand_scale(mult = c(0, 0.1))
  ) +
  facet_grid(
    cols = vars(site), 
    scales = "free", 
    labeller = labeller(site = site_labels)
  ) + 
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 4.5
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )


total_anaerobe

```

## Statistics reported in Supplementary Table S2 (total anaerobe)

### total anaerobe - CREC

**Normality:** all approximately normal

```{r message=FALSE}

total_anaerobe %>% 
  filter(site == "CREC", amend == "U") %>%
  group_by(site, till, amend) %>% 
  summarise(
    normality_p = shapiro.test(anaerobe_copies_per_g)$p.value
  ) %>%
  arrange(normality_p)

```

**Variances:** Variances are barely equal

```{r}
bartlett.test(
  anaerobe_copies_per_g ~ till,
  data = total_anaerobe %>% 
  filter(site == "CREC", amend == "U")
)
```

**Statistics:** ANOVA

```{r}

crec_anaer_aov <- 
  aov(
    anaerobe_copies_per_g ~ till,
    data = total_anaerobe %>% 
      filter(site == "CREC", amend == "U")
  ) 

crec_anaer_aov %>% summary()

TukeyHSD(crec_anaer_aov)
```

### total anaerobe -  SM

**Normality:** all approximately normal

```{r message=FALSE}

total_anaerobe %>% 
  filter(site == "SM", amend == "U") %>%
  group_by(site, till, amend) %>% 
  summarise(
    normality_p = shapiro.test(anaerobe_copies_per_g)$p.value
  ) %>%
  arrange(normality_p)

```

**Variances:** Variances are equal

```{r}
bartlett.test(
  anaerobe_copies_per_g ~ till,
  data = total_anaerobe %>% 
  filter(site == "SM", amend == "U")
)
```

**Statistics:** ANOVA 

```{r}

sm_anaer_aov <-
  aov(
    anaerobe_copies_per_g ~ till,
    data = total_anaerobe %>% 
      filter(site == "SM", amend == "U")
  ) 

sm_anaer_aov %>% summary()

TukeyHSD(sm_anaer_aov)

```

### total anaerobe - WO

**Normality:** 

```{r message=FALSE}

total_anaerobe %>% 
  filter(site == "WO", amend == "U", till != "UN") %>%
  group_by(site, till, amend) %>% 
  summarise(
    normality_p = shapiro.test(sqrt(anaerobe_copies_per_g))$p.value
  ) %>%
  arrange(normality_p)

```
Assume normality for WO-UN

**Variances:** Equal

```{r}

bartlett.test(
    anaerobe_copies_per_g ~ till,
  data = total_anaerobe %>% 
  filter(site == "WO", amend == "U")
)

```

**Statistics:** ANOVA

```{r}

wo_aov <- 
aov(  
  anaerobe_copies_per_g ~ till,
  data = total_anaerobe %>% 
    filter(site == "WO", amend == "U")
)

summary(wo_aov)

TukeyHSD(wo_aov)

```

# Tillage - Carrington only

## Figure - stacked bar (unpublished)

```{r message=FALSE}

all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n()),
  ) %>% 
  filter(site == "CREC", amend == "U") %>%
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  left_join(errors, by = c("site", "till", "amend", "landscape_position")) %>% 
  ggplot(aes(x = till, fill = target_gene)) + 
  geom_col(aes(y = avg_copies_per_g), position = "stack", color = "black") +
  geom_errorbar(
    aes(y = sum_copies, ymin = sum_copies, ymax = sum_copies + sum_se)
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(labels = scientific) +
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 2
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )

```

## Figure - faceted (unpublished)

```{r message=FALSE}

all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n())
  ) %>% 
  filter(site == "CREC", amend == "U") %>% 
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  ggplot(aes(y = avg_copies_per_g, x = till, fill = target_gene)) + 
  geom_col(position = "dodge", color = "black") +
  geom_errorbar(
    aes(
      ymin = avg_copies_per_g,
      ymax = avg_copies_per_g + se_copies_per_g
    ),
    position = "dodge"
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(labels = scientific) +
  facet_grid(
    rows = vars(target_gene), 
    scales = "free",
    labeller = labeller(site = site_labels)
  ) + 
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 0.5,
    legend.position = "none",
    strip.text.y = element_text(face = "italic")
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )

```

# Absolute Abundance: Landscape Position 

## Supplementary Figure S5b - faceted

```{r message=FALSE}

all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n())
  ) %>% 
  filter(site == "GR") %>% 
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  ggplot(aes(y = avg_copies_per_g, x = landscape_position, fill = target_gene)) + 
  geom_col(position = "dodge", color = "black") +
  geom_errorbar(
    aes(
      ymin = avg_copies_per_g,
      ymax = avg_copies_per_g + se_copies_per_g
    ),
    position = "dodge"
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(labels = scientific) +
  facet_grid(
    rows = vars(target_gene), 
    scales = "free"
  ) + 
  theme_bw() + 
  my_theme() +
  theme(
    legend.position = "none",
    strip.text.y = element_text(face = "italic", size = 14),
    aspect.ratio = 2,
    axis.text.x = element_text(angle = 45, hjust = 1, size = 13),
    axis.text.y = element_text(size = 13),
    axis.title.y = element_text(size = 16)
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )

```


## Supplementary Figure S5a - stacked bars

```{r message=FALSE}

all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n()),
  ) %>% 
  filter(site == "GR") %>%
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  left_join(errors, by = c("site", "till", "amend", "landscape_position")) %>% 
  ggplot(aes(x = landscape_position, fill = target_gene)) + 
  geom_col(aes(y = avg_copies_per_g), position = "stack", color = "black") +
  geom_errorbar(
    aes(y = sum_copies, ymin = sum_copies, ymax = sum_copies + sum_se)
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(labels = scientific) +
  scale_x_discrete(
    labels = c("C" = "Channel", "F" = "Footslope", "S" = "Shoulder")
  ) +
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 2,
    axis.text.x = element_text(angle = 45, hjust = 1, size = 13),
    axis.text.y = element_text(size = 13),
    axis.title.y = element_text(size = 16),
    legend.text = element_text(size = 13)
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )


```



# Absolute Abundance - Manure amendments at Carrington

## Figure 3a - stacked bars

```{r message=FALSE}

all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n()),
  ) %>% 
  filter(site == "CREC", till %in% c("CT", "NT")) %>%
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  left_join(errors, by = c("site", "till", "amend", "landscape_position")) %>% 
  ggplot(aes(x = amend, fill = target_gene)) + 
  geom_col(aes(y = avg_copies_per_g), position = "stack", color = "black") +
  geom_errorbar(
    aes(y = sum_copies, ymin = sum_copies, ymax = sum_copies + sum_se)
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(labels = scientific) +
  facet_grid(
    cols = vars(till), 
    scales = "free", 
    labeller = labeller(site = site_labels)
  ) + 
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 3
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )

```

## Statistical Tests reported in Supplementary Table S3 (Total anaerobe)

### total anaerobe - manure effect

#### Check normality

```{r message=FALSE}

total_anaerobe %>% 
  filter(site == "CREC") %>% 
  group_by(site, till, amend) %>% 
  summarise(
    normality_p = shapiro.test(anaerobe_copies_per_g)$p.value
  ) %>% 
  arrange(normality_p)

```
#### Test for equal variances

```{r}

leveneTest(
  anaerobe_copies_per_g~amend*till,
  data = total_anaerobe %>% filter(site == "CREC", till %in% c("CT", "NT"))
)

```

#### test for differences in amended vs. unamended soils 
Will use t-test for differences between amended and unamended

**CT**

```{r}

t.test(
  x = 
    total_anaerobe %>% 
    filter(site == "CREC", till == "CT", amend == "A") %>% 
    pull(anaerobe_copies_per_g),
  y = 
    total_anaerobe %>% 
    filter(site == "CREC", till == "CT", amend == "U") %>% 
    pull(anaerobe_copies_per_g),
  alternative = c("greater")
)

```

**NT**

```{r}

t.test(
  x = 
    total_anaerobe %>% 
    filter(site == "CREC", till == "NT", amend == "A") %>% 
    pull(anaerobe_copies_per_g),
  y = 
    total_anaerobe %>% 
    filter(site == "CREC", till == "NT", amend == "U") %>% 
    pull(anaerobe_copies_per_g),
  alternative = c("greater") #this really depends! 
)

```

## Figure 3b - faceted

```{r message=FALSE}

all_ddpcr_data %>% 
  group_by(site, till, amend, landscape_position, target_gene) %>% 
  summarise(
    avg_copies_per_g = mean(copies_per_g, na.rm = TRUE),
    se_copies_per_g = sd(copies_per_g, na.rm = TRUE) / sqrt(n())
  ) %>% 
  filter(site == "CREC", till %in% c("CT", "NT")) %>% 
  mutate(
    across(
      target_gene, 
      ~factor(
        ., 
        levels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA"),
        labels = c("nirK", "nirS", "gltA", "dsrAB", "mcrA")
      )
    )
  ) %>% 
  ggplot(aes(y = avg_copies_per_g, x = amend, fill = target_gene)) + 
  geom_col(position = "dodge", color = "black") +
  geom_errorbar(
    aes(
      ymin = avg_copies_per_g,
      ymax = avg_copies_per_g + se_copies_per_g
    ),
    position = "dodge"
  ) +
  scale_fill_brewer(palette = "RdYlBu") +
  scale_y_continuous(labels = scientific) +
  facet_grid(
    cols = vars(till), 
    rows = vars(target_gene), 
    scales = "free",
    labeller = labeller(site = site_labels)
  ) + 
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 1,
    legend.position = "none",
    strip.text.y = element_text(face = "italic")
  ) +
  labs(
    y = "Copies per g soil",
    x = NULL
  )

```

## Statistical Tests reported in Supplementary Table S3 

### nirK

**Normality:** Approximately normal

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(
    site == "CREC", 
    till %in% c("CT", "NT"), 
    target_gene == "nirK"
  ) %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(log(copies_per_g))$p.value
  ) %>%
  arrange(target_gene, normality_p)

```

#### CT

**Variances**

```{r}

bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "nirK", till == "CT")
) 

```


**Statistics:** t-test
```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "A", 
      target_gene == "nirK"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "U",
      target_gene == "nirK"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = TRUE
)

```

#### NT

**Variances**

```{r}

bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "nirK", till == "NT")
) 

```

**Statistics:** t-test

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "A", 
      target_gene == "nirK"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "U",
      target_gene == "nirK"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = TRUE
)

```

### nirS

**Normality**

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(
    site == "CREC", 
    till %in% c("CT", "NT"), 
    target_gene == "nirS"
  ) %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(target_gene, normality_p)

```

#### CT

**Variances**

```{r}
bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "nirS", till == "CT")
) 

```


**Statistics:** t-test

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "A", 
      target_gene == "nirS"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "U",
      target_gene == "nirS"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = FALSE
)

```

#### NT

**Variances**

```{r}

bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "nirS", till == "NT")
) 

```


**Statistics**

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "A", 
      target_gene == "nirS"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "U",
      target_gene == "nirS"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = TRUE
)

```

### gltA

**Normality**

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(
    site == "CREC", 
    till %in% c("CT", "NT"), 
    target_gene == "gltA"
  ) %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(target_gene, normality_p)

```

#### CT

**Variances**
```{r}
bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "gltA", till == "CT")
) 

```


**Statistics:** t-test

```{r}
t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "A", 
      target_gene == "gltA"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "U",
      target_gene == "gltA"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = TRUE
)
```

#### NT

**Variances**

```{r}

bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "gltA", till == "NT")
) 

```

**Statistics:** t-test

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "A", 
      target_gene == "gltA"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "U",
      target_gene == "gltA"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = TRUE
)

```

### dsrAB

**Normality:** normal

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(
    site == "CREC", 
    till %in% c("CT", "NT"), 
    target_gene == "dsrAB"
  ) %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(log(copies_per_g))$p.value
  ) %>%
  arrange(target_gene, normality_p)

```

#### CT

**Variances**

```{r}

bartlett.test( 
  log(copies_per_g) ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "dsrAB", till == "CT")
) 

```

**Statistics:** t-test

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "A", 
      target_gene == "dsrAB"
    ) %>% 
    mutate(
      log_copies_per_g = log(copies_per_g)
    ) %>% 
    pull(log_copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "U",
      target_gene == "dsrAB"
    ) %>% 
    mutate(
      log_copies_per_g = log(copies_per_g)
    ) %>% 
    pull(log_copies_per_g),
  alternative = c("greater"),
  var.equal = TRUE
)

```

#### NT

**Variances**

```{r}

bartlett.test( 
  log(copies_per_g) ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "dsrAB", till == "NT")
) 

```

**Statistics:** t-test

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "A", 
      target_gene == "dsrAB"
    ) %>% 
    mutate(
      log_copies_per_g = log(copies_per_g)
    ) %>% 
    pull(log_copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "U",
      target_gene == "dsrAB"
    ) %>% 
    mutate(
      log_copies_per_g = log(copies_per_g)
    ) %>% 
    pull(log_copies_per_g),
  alternative = c("greater"),
  var.equal = TRUE
)

```

### mcrA

**Normality:** Approximately normal

```{r message=FALSE}

all_ddpcr_data %>% 
  filter(
    site == "CREC", 
    till %in% c("CT", "NT"), 
    target_gene == "mcrA"
  ) %>%
  group_by(site, till, amend, target_gene) %>% 
  summarise(
    normality_p = shapiro.test(copies_per_g)$p.value
  ) %>%
  arrange(target_gene, normality_p)

```

#### CT

**Variances:** unequal

```{r}

bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "mcrA", till == "CT")
) 

```

**Statistics:** t-test

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "A", 
      target_gene == "mcrA"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "CT", 
      amend == "U",
      target_gene == "mcrA"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = FALSE
)

```

#### NT

**Variances**

```{r}

bartlett.test( 
  copies_per_g ~ amend,
  data = 
    all_ddpcr_data %>% 
    filter(site == "CREC", target_gene == "mcrA", till == "NT")
) 

```
**Statistics:** t-test

```{r}

t.test(
  x = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "A", 
      target_gene == "mcrA"
    ) %>% 
    pull(copies_per_g),
  y = 
    all_ddpcr_data %>% 
    filter(
      site == "CREC", 
      till == "NT", 
      amend == "U",
      target_gene == "mcrA"
    ) %>% 
    pull(copies_per_g),
  alternative = c("greater"),
  var.equal = FALSE
)

```

# PCA of Anaerobe Abundance

## Create tibble for PCA

```{r}

anaerobe_pca_data <-
  all_ddpcr_data_w_outliers %>% 
  group_by(site, till, amend, rep,landscape_position, target_gene) %>% 
  summarise(
    copies_per_g = mean(copies_per_g)
  ) %>% 
  pivot_wider(
    id_cols = c("site", "till", "amend", "rep", "landscape_position"),
    names_from = "target_gene", 
    values_from = "copies_per_g"
  ) %>% 
  drop_na(dsrAB, gltA, mcrA, nirK) %>% 
  ungroup() %>% 
  # Values below were removed because they were outliers
  filter(
    !(site == "WO" & till == "UN" & rep == 2),
    !(site == "CREC" & till == "CT" & amend == "A"& rep == 1),
    !(site == "CREC" & till == "CT" & amend == "A"& rep == 2),
    !(site == "SM" & till == "NT" & amend == "U"& rep == 1),
  ) %>% 
  mutate(
    log_dsrAB = log(dsrAB),
    log_gltA = log(gltA),
    log_mcrA = log(mcrA + (0.5*23957.38)),
    sqrt_nirK = sqrt(nirK),
    log_nirS = log(nirS),
  ) 


```

## Check normality

```{r}
anaerobe_pca_data %>% 
  ggplot() +
  geom_histogram(aes(x = log(dsrAB)))


anaerobe_pca_data %>% 
  ggplot() +
  geom_histogram(aes(x = log(gltA)))

anaerobe_pca_data %>% 
  ggplot() +
  geom_histogram(aes(x = log(mcrA+ (0.5*23957.38))))

anaerobe_pca_data %>% 
  ggplot() +
  geom_histogram(aes(x = sqrt(nirK)))

anaerobe_pca_data %>% 
  ggplot() +
  geom_histogram(aes(x = log(nirS)))
```



## Confirm outliers removed 

Code originally used to identify outliers

```{r}
grubbs.test(anaerobe_pca_data$log_dsrAB)

grubbs.test(anaerobe_pca_data$log_gltA) #there is an outlier (CREC-CT-A-1 and CT-A-2)

grubbs.test(anaerobe_pca_data$log_mcrA) 

grubbs.test(anaerobe_pca_data$sqrt_nirK) #there is an outlier (WO-UN-U-2)

grubbs.test(anaerobe_pca_data$log_nirS) #there is an outlier (SM-NT-U-1)


```


## Perform PCA 
```{r}
pca_anaerobes <- 
  prcomp(
    ~log_dsrAB + log_gltA + log_mcrA + sqrt_nirK + log_nirS, 
    data = anaerobe_pca_data, 
    scale = TRUE
  )



```
### Plot - Supplementary Figure S2

```{r}
autoplot(
  pca_anaerobes,
  loadings = TRUE,
  loadings.label = TRUE,
) +
  scale_x_reverse() + 
  theme_bw() + 
  theme(
    aspect.ratio = 1,
    axis.title = element_text(size = 16),
    axis.text = element_text(size = 14)
  )

```

### Save PC1 values
NOTE: anaerobes_by_pca was saved as a csv: `anaerobe_pca.csv`

```{r}
pc1_values <- pca_anaerobes$x %>% as_tibble() %>%  pull(PC1)


anaerobes_by_pca <- 
  anaerobe_pca_data %>% 
  add_column(pc1_values) %>% 
  mutate(
    pc1_values_trans = 1 * pc1_values
  )

```

# PC1 vs. tillage 

## Supplementary Figure S3

```{r}
anaerobes_by_pca %>% 
  group_by(site, till, landscape_position, amend) %>% 
  summarise(
    avg_pc1 = mean(pc1_values_trans, na.rm = TRUE),
    se_pc1 = sd(pc1_values_trans, na.rm = TRUE) / sqrt(n()),
  ) %>% 
  filter(site != "GR", till != "MT", amend == "U") %>%
  ggplot() +
  geom_pointrange(
    aes(
      x = till, 
      y = avg_pc1, 
      ymin = avg_pc1 - se_pc1, 
      ymax = avg_pc1 + se_pc1,
    )
  ) + 
  scale_y_reverse() +
  expand_limits(y = -3.5) +
  facet_grid(
    cols = vars(site), 
    scales = "free", 
    labeller = labeller(site = site_labels)
  ) + 
  my_theme() + 
  theme(
    aspect.ratio = 1.5
  ) +
  labs(
    y = "PC1 - Anaerobe Abundances",
    x = NULL
  )

```

## Statistics - values reported in Supplementary Table S2

*Normality* 
```{r}
anaerobes_by_pca %>% 
  group_by(site, till, amend) %>% 
  mutate(
    count = n()
  ) %>% 
  filter(count > 2) %>% 
  summarise(
    normality_p = shapiro.test(pc1_values_trans)$p.value
  ) %>%
  arrange(normality_p)
```
All groups w/ more than 3 observations are normally distributed. 

*Variances*

```{r}

bartlett.test(
   pc1_values_trans ~ interaction(till, site),
  data = 
    anaerobes_by_pca %>% 
    filter(site != "GR", till != "MT", amend == "U") 
  
)

```
### CREC

```{r}

crec_pc_anaerobe_aov <- 
  aov(
      pc1_values_trans ~ till,
      data = 
        anaerobes_by_pca %>% 
        filter(site == "CREC", amend == "U", till != "MT") 
)

crec_pc_anaerobe_aov %>% summary()

TukeyHSD(crec_pc_anaerobe_aov)

```


### SM

```{r}
sm_pc_anaerobe_aov <- 
  aov(
      pc1_values_trans ~ till,
      data = 
        anaerobes_by_pca %>% 
        filter(site == "SM", amend == "U", till != "MT") 
)

sm_pc_anaerobe_aov %>% summary()

```



### WO

```{r}
wo_pc_anaerobe_aov <- 
  aov(
      pc1_values_trans ~ till,
      data = 
        anaerobes_by_pca %>% 
        filter(site == "WO", amend == "U", till != "MT") 
)

wo_pc_anaerobe_aov %>% summary()


TukeyHSD(wo_pc_anaerobe_aov)

```
# PC1 Anaerobe Abundances vs. manure amendments

## Supplementary Figure S4

```{r}
anaerobes_by_pca %>% 
  group_by(site, till, amend, landscape_position) %>% 
  summarise(
    avg_pc = mean(pc1_values_trans, na.rm = TRUE),
    se_pc = sd(pc1_values_trans, na.rm = TRUE) / sqrt(n()),
  ) %>% 
  filter(site == "CREC", till %in% c("CT", "NT")) %>% 
  ggplot(aes(x = amend)) + 
  geom_pointrange(aes(ymin = avg_pc - se_pc, ymax = avg_pc + se_pc, y = avg_pc)) +
  scale_y_reverse() +
  expand_limits(y = -2.5) +
  facet_grid(
    cols = vars(till), 
    scales = "free", 
    labeller = labeller(site = site_labels)
  ) + 
  theme_bw() + 
  my_theme() +
  theme(
    aspect.ratio = 1.5
  ) +
  labs(
    y = "PC1 - Anaerobe Abundances",
    x = NULL
  )

```

## Statistics - Reported in Supplementary Table S3 

### CT

Not enough PC1 observations for CT-A to perform t-test.

### NT

*Check normality*

Data is normal. 
```{r message=FALSE}

anaerobes_by_pca %>% 
  filter(site == "CREC", till == "NT") %>% 
  group_by(site, till, amend) %>%
  summarise(
    normality_p = shapiro.test(pc1_values)$p.value
  ) %>% 
  arrange(normality_p)

```

*Test for equal variances*

```{r}

bartlett.test( 
  pc1_values ~ amend,
  data = 
      anaerobes_by_pca %>% 
    filter(site == "CREC", till == "NT")
)

```

Variances are equal. 


*t-test*

```{r}
t.test(
  x = 
    anaerobes_by_pca %>% 
    filter(site == "CREC", till == "NT", amend == "A") %>% 
    pull(pc1_values),
  y = 
    anaerobes_by_pca %>% 
    filter(site == "CREC", till == "NT", amend == "U") %>% 
    pull(pc1_values),
  var.equal = TRUE,
  alternative = c("less") #more negative = more anaerobes
)

```


